### PR TITLE
feat: add structured object output support for task API

### PR DESCRIPTION
**Pull request by WashizuRyo**
## Summary
Implements a `buildObject` function that assembles structured JSON objects from task generation outputs based on schema definitions and property mappings. The task API response now supports two output types: `passthrough` (default, existing behavior) and `object` (new structured output), determined by the `endNodeOutput.format` discriminator.

Note: Retrieving structured outputs via the SDK is not yet supported. This will be addressed in a follow-up PR.

## Related Issue
N/A

## Changes
- Added `buildObject` function in `packages/giselle/src/tasks/build-object.ts` that resolves property mappings against completed generation outputs to construct typed JSON objects, supporting nested objects, arrays, and type coercion (number, boolean, string)
- Added comprehensive test suite with 633 lines covering valid mappings (flat, nested, array, multi-source, type coercion) and invalid/edge cases (missing nodes, missing outputs, unresolvable paths)
- Refactored the task GET API route to use a discriminated `outputType` (`"passthrough"` | `"object"`) response shape, with `passthrough` as the default preserving backward compatibility
- Simplified the `ApiStepItem` type by removing redundant union variant
- Exported `buildObject` from `@giselles-ai/giselle` package

## Testing
Comprehensive unit tests in `build-object.test.ts` covering:
- Single and multi-property mapping
- Nested object and array resolution
- Type coercion (number, boolean)
- Missing generation, missing output, and unresolvable path edge cases

## Other Information
N/A

---

> [!NOTE]
> **Medium Risk**
> Changes the `GET /api/apps/:appId/tasks/:taskId` JSON shape (adds `outputType` and conditionally replaces `outputs` with `output`) and introduces new object-building logic from generation outputs, which could impact strict API consumers and edge-case output mapping.
> 
> **Overview**
> **Adds structured output mode to the task GET API.** When `includeGenerations=1`, the response now returns a discriminated `task` result: `outputType: "passthrough"` with existing `outputs`, or `outputType: "object"` with a computed JSON `output` (no `outputs` duplication).
> 
> **Introduces engine-level structured object construction.** Adds and exports `buildObject` in `@giselles-ai/giselle`, plus a comprehensive test suite, to build the `object` output by resolving schema/property mappings against completed generations (with best-effort coercion and omission of unresolvable fields).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 996babf62d47d5a43e84ae5fb78c75c0b6b8a4fb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added an object-structured output format for task results alongside existing passthrough, with responses now indicating the output type.
  * Task results can now return a constructed object by resolving generation data into nested properties; passthrough remains the default for compatibility.
  * Nested object fields are omitted when child properties cannot be resolved.

* **Tests**
  * New test coverage for object construction, type coercion, array handling, path resolution, and failure/omission cases.

### CODE REVIEW

**Comment by WashizuRyo** [comment]
I'm checking agents' feedback now.

File: packages/giselle/src/tasks/build-object.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	switch (targetSchema.type) {
+		case "object":
+			return value !== null &&
+				typeof value === "object" &&
+				!Array.isArray(value)
+				? value
+				: undefined;
+		case "array":
+			return Array.isArray(value) ? value : undefined;
+		case "number": {
```
[the remark this answers is not in the document]
Comment (WashizuRyo) [reply packages/giselle/src/tasks/build-object.ts:87]: https://github.com/giselles-ai/giselle/issues/2770

I'll address this in a separate PR.

**Comment by WashizuRyo** [comment]
Done ✅ https://github.com/giselles-ai/giselle/pull/2769#issuecomment-3970684230

**Review by toyamarinyon — APPROVED** [review]
